src/clj/harja/palvelin/palvelut/materiaalit.clj

```diff
@@ -211,22 +211,23 @@
         (q/hae-suolamateriaalit db)))
 
 (defn luo-suolatoteuma [db user urakka-id sopimus-id toteuma]
-  (let [t (toteumat/luo-toteuma<!
-           db urakka-id sopimus-id
-           (:alkanut toteuma) (:alkanut toteuma)
-           "kokonaishintainen"
-           (:id user) "" ""
-           (:lisatieto toteuma)
-           nil nil nil nil nil nil nil
-           "harja-ui")]
-    (toteumat/luo-toteuma-materiaali<!
-     db (:id t) (:id (:materiaali toteuma))
-     (:maara toteuma) (:id user))))
+  (let [t (toteumat-q/luo-toteuma<!
+            db urakka-id sopimus-id
+            (:alkanut toteuma) (:alkanut toteuma)
+            "kokonaishintainen"
+            (:id user) "" ""
+            (:lisatieto toteuma)
+            nil nil nil nil nil nil nil
+            "harja-ui")]
```

Comment:
Voisi toki olla näin. Toisaalta minusta tuntuu, että pääsääntöisesti frontin palvelut ja API:n käyttämät on eroteltu toisistaan.